api/filter_outputs.go

```diff
@@ -0,0 +1,28 @@
+package api
+
+import (
+	"net/http"
+
+	"github.com/pkg/errors"
+)
+
+func (api *API) createFilterOutput(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+
+	var request createFilterOutputRequest
+	if err := api.ParseRequest(r.Body, &request); err != nil {
+		api.respond.Error(ctx, w, http.StatusBadRequest, errors.Wrap(err, "parse request"))
+		return
+	}
+	if err := api.validate.Struct(&request); err != nil {
+		api.respond.Error(ctx, w, http.StatusBadRequest, errors.Wrap(err, "validate request fields"))
+		return
+	}
+
+	if err := api.store.CreateFilterOutput(ctx, request.Downloads); err != nil {
+		api.respond.Error(ctx, w, http.StatusBadRequest, errors.Wrap(err, "upsert filter output"))
```

Comment:
Please can you make the error messages more human readable i.e `failed to create filter output` rather than just `upsert filter output`. Also we don't want to be returning database implementation details to the user in the error response, see the existing endpoints for how to tailor a separate message for the caller to what gets logged. You might want to think about adding log data to your errors as well.

Reply:
done